fish-rust/src/tests/common.rs

```diff
@@ -0,0 +1,75 @@
+use crate::common::{
+    escape_string, unescape_string, EscapeFlags, EscapeStringStyle, UnescapeStringStyle,
+};
+use crate::future_feature_flags::future_feature_flags_init;
+use crate::wchar::WString;
+use crate::wchar::L;
+use libc::{setlocale, LC_ALL};
+use rand::random;
+use std::ffi::CString;
+
+/// The number of tests to run.
+const ESCAPE_TEST_COUNT: usize = 100_000;
+/// The average length of strings to unescape.
+const ESCAPE_TEST_LENGTH: usize = 100;
+/// The highest character number of character to try and escape.
+const ESCAPE_TEST_CHAR: usize = 4000;
+
+macro_rules! test_escape_style {
+    ($name:ident: $escape_style:expr, $unescape_style:expr) => {
+        #[test]
+        pub fn $name() {
+            future_feature_flags_init();
+            unsafe {
+                let utf8_locale = CString::new("en_US.UTF-8").unwrap();
+                setlocale(LC_ALL, utf8_locale.as_ptr());
+            }
```

Comment:
This is a mess. `feature_flags` should just init automatically, and the tests default to a `C`-locale which results in strings of `????`. I am not sure what's the more idiomatic way to deal with the locale issue, there's gotta be a config _somewhere_

Reply:
The tests also take ~1min for var and url, and ~10s for script, can be sped up by ~20s by printf-compat @ opt-level=3

Reply:
Some pointers for how to add a `setup()`: https://www.infinyon.com/blog/2021/04/rust-custom-test-harness/, https://rust-lang.github.io/rfcs/2318-custom-test-frameworks.html, https://github.com/infinyon/fluvio/blob/master/crates/fluvio-test/src/main.rs, https://github.com/infinyon/fluvio/blob/6635b8aae20df796604d60d2706289984bcc8da4/tests/runner/src/fluvio-integration-derive/src/lib.rs, in other words not trivial